python/pyspark/sql/readwriter.py

```diff
@@ -327,6 +334,8 @@ def csv(self, path, schema=None, sep=None, encoding=None, quote=None, escape=Non
                         character. By default (None), it is disabled.
         :param header: uses the first line as names of columns. If None is set, it uses the
                        default value, ``false``.
+        :param inferSchema: infers the input schema automatically from data. It requires one extra
```

Comment:
Yes, it will be off (so I wrote some comments above). However, thing is about `header`. If `inferSchema` option is removed and it is only enabled when the schema is set, there would be no way to not infer schema (meaning all string columns) but only use column names in the header.